lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -93,8 +93,11 @@ def parse(document, options = {})
           attributes['id'] = primary_data['id'] if primary_data['id']
           relationships = primary_data['relationships'] || {}
 
-          filter_fields(attributes, options)
-          filter_fields(relationships, options)
+          attributes = transform_keys(attributes, options)
+          relationships = transform_keys(relationships, options)
```

Comment:
> , maybe I should be calling transform_keys on the only/except data as well

No, that's used internally by us (Rails), so once we've done the key transform on the request data, then the `only` options should be using the *internal* casing, which in Rails is underscore